internal/storage/v1/elasticsearch/spanstore/writer.go

```diff
@@ -148,3 +170,22 @@ func (s *SpanWriter) writeService(indexName string, jsonSpan *dbmodel.Span) {
 func (s *SpanWriter) writeSpan(indexName string, jsonSpan *dbmodel.Span) {
 	s.client().Index().Index(indexName).Type(spanType).BodyJson(&jsonSpan).Add()
 }
+
+func (s *SpanWriter) splitElevatedTags(keyValues []dbmodel.KeyValue) ([]dbmodel.KeyValue, map[string]any) {
+	var tagsMap map[string]any
```

Comment:
nit: I think we should short circuit here
 ```suggestion
	if (!s.allTagsAsFields || len(s.tagKeysAsFields) == 0) {
		return
	}
	var tagsMap map[string]any
```